### PR TITLE
[DOPS-3156]: automatize runtime upgrades

### PR DESCRIPTION
**Pull request by f33r0**
# Task

[DOPS-3156]: automatize runtime upgrades

## Changes

1. Modified script Dockerfile

[DOPS-3156]: https://soramitsu.atlassian.net/browse/DOPS-3156

### CODE REVIEW

File: misc/runtime_upgrade/Dockerfile [hunk 1]
```diff
@@ -1,15 +1,23 @@
...
 
-RUN pip install --no-cache-dir -r requirements.txt
+USER appuser
 
-COPY main.py .
-
-USER nonroot
-
-ENTRYPOINT  ["python", "main.py"]
+CMD ["python", "main.py"]
```
Comment (Alexey-N-Chernyshov) [inline misc/runtime_upgrade/Dockerfile:23]:
```suggestion
CMD ["python", "main.py"]

```

File: runtime/src/migrations.rs [hunk 2]
```diff
@@ -29,68 +29,9 @@
...
-            None,
-        ) {
-            error!("Failed to register KUSD asset, error: {:?}", err);
-            return <T as frame_system::Config>::DbWeight::get().reads(1);
-        }
-        <T as frame_system::Config>::BlockWeights::get().max_block
-    }
-}
 
+#[cfg(feature = "ready-to-test")] // kensetsu
```
Comment (Alexey-N-Chernyshov) [inline runtime/src/migrations.rs:33]: #998 missing, please merge `master` when the PR is merged.

**Review by Alexey-N-Chernyshov — COMMENTED** [review]
Do we need `framenode_runtime.compact.compressed.wasm` binary file in repo? May be add it to `gitignore`?

File: Jenkinsfile [hunk 3]
```diff
@@ -1,4 +1,4 @@
-@Library('jenkins-library') _
+@Library('jenkins-library@feature/dops-3156/runtime_upgrade') _
```
Comment (safinsaf) [inline Jenkinsfile:1]: Pls do not forget to remove before merge

**Review by safinsaf — APPROVED** [review]

File: misc/runtime_upgrade/Dockerfile [hunk 4]
```diff
@@ -1,15 +1,23 @@
-FROM python:3.11-slim
+FROM python:3.11-alpine3.19
 
 WORKDIR /app
+COPY ./misc/runtime_upgrade/main.py \
```
Comment (wpi86) [inline misc/runtime_upgrade/Dockerfile:4]: It's more secure if we have immutable scripts.

**Review by wpi86 — CHANGES_REQUESTED** [review]

**Review by C4tWithShell — APPROVED** [review]

**Review by Cre-eD — APPROVED** [review]

**Review by wpi86 — CHANGES_REQUESTED** [review]
fix script permissions

**Review by RamilMus — APPROVED** [review]

**Review by wpi86 — APPROVED** [review]

**Review by Alexey-N-Chernyshov — APPROVED** [review]

**Pull request merged**